pkg/tasks/tasks.go

```diff
@@ -102,25 +102,40 @@ type TaskGroup struct {
 	tasks []*TaskSpec
 }
 
-func NewTaskSpec(name string, task Task) *TaskSpec {
+// NewTaskSpec creates a task that doesn't have a UWM component as a target
+func NewTaskSpec(target string, task Task) *TaskSpec {
 	return &TaskSpec{
-		Name: name,
-		Task: task,
+		Name:       fmt.Sprintf("Updating%s", target),
+		UWMRelated: false,
+		Task:       task,
+	}
+}
+
+// NewUWMTaskSpec creates a task that has UWM components as a target
+func NewUWMTaskSpec(target string, task Task) *TaskSpec {
```

Comment:
Maybe we want to keep `tasks.go` UWM stuff-agnostic? I can try another approach

Reply:
I think it's fine as long as we mark whatever comes from UWM as such, as you're stating there.

Reply:
We don't use `tasks.go` anywhere else so there is generally no need to keep things decoupled. One observation though: `operator.go` is really the only component that needs to know which part (platform or UWM) a specific task updates. `NewUWMTaskSpec` is really just a convenience method that sets `UWMRelated: true` and a specfic name prefix.

Couldn't all this be handled from `operator.go`? We could leave `tasks.go` unchanged and have `operator.go` gain a convenience method that adds a UWM specific prefix to a task. Based on that prefix we can also alter the reporting to differentiate between platform and UWM errors.

Reply:
Sure, here is a version https://github.com/openshift/cluster-monitoring-operator/compare/master...machine424:cluster-monitoring-operator:uwm-tasks-2?expand=1

The only concern I have is that using/relying on prefixes always seem hacky/error-prone even though we control the whole chain...

I think we can reduce the functions composition `tasks.NewTaskSpec(TaskName("...` e.g., by wrapping the whole in a function, sth like https://github.com/openshift/cluster-monitoring-operator/compare/master...machine424:cluster-monitoring-operator:uwm-tasks-3?expand=1

WDYT?

Reply:
@jan--f (in case this went unnoticed)